### DISCUSSION TITLE
installation of popper.js into BootstrapVue

### DISCUSSION THREAD

**Discussion by darkomenx** [bootstrap-vue/bootstrap-vue#6052] [Q&A]
### Documentation issue

- [ ] Reporting a typo
- [ ] Reporting a documentation bug
- [X] Documentation improvement
- [ ] Documentation feedback

### Is there a specific documentation page you are reporting?

https://bootstrap-vue.org/docs

### Additional context or description

The documentation tell that BootstrapVue use the library popper.js but there is nothing about how integrate that into project vuejs which use BootstrapVue. I search information about that but I found nothing and I can't use specific component that use this library. 

I just want to use your library in full mode for all components available :)

**Comment by darkomenx** [comment]
I'm also try this : 

import all plugin into bootstrap-vue module like this : 

`import { BootstrapVue, IconsPlugin, BootstrapVueIcons, DropdownPlugin, PopoverPlugin, ProgressPlugin, AlertPlugin, AspectPlugin, AvatarPlugin, BadgePlugin } from 'bootstrap-vue'
`

and after connect those plugin imported to Vue framework runtime like this : 

```
Vue.use(BootstrapVue)

Vue.use(IconsPlugin)

Vue.use(BootstrapVueIcons)

Vue.use(DropdownPlugin)

Vue.use(PopoverPlugin)

Vue.use(ProgressPlugin)

Vue.use(AlertPlugin)

Vue.use(AspectPlugin)

Vue.use(AvatarPlugin)

Vue.use(BadgePlugin)
```

But I can't view correct dropdown or popper component

**Comment by Hiws** [comment]
These two lines will import **all** components and icons.
```js
Vue.use(BootstrapVue)
Vue.use(IconsPlugin)
```
So all the other imports aren't necessary and might be increasing the bundle size for no reason.
This *might* also the reason your dropdown isn't working.

Did you remember to import both Bootstrap *and* BootstrapVue CSS/SCSS into your project?

**Comment by darkomenx** [comment]
> So all the other imports aren't necessary and might be increasing the bundle size for no reason.
> This might also the reason your dropdown isn't working.

OK I log that. But It is not the origin of issue because I begin start with those only 2 lines and not all others (this is just a test). Issue's origin is not that.

> Did you remember to import both Bootstrap and BootstrapVue CSS/SCSS into your project?

I use a main.scss file that imported into my components vuejs that contain this : 

```
@import 'node_modules/bootstrap/scss/bootstrap.scss';
@import 'node_modules/bootstrap/scss/mixins';
@import 'node_modules/bootstrap/scss/dropdown';
@import 'node_modules/bootstrap/scss/popover';
@import 'node_modules/bootstrap/scss/variables';
@import 'node_modules/bootstrap/scss/utilities';
@import 'node_modules/bootstrap-vue/src/index.scss';
```

or I try with this line :  

```
@import 'node_modules/bootstrap/scss/bootstrap.scss';
@import 'node_modules/bootstrap-vue/src/index.scss';
```

is it correct ?

Thank you for your response and your library and welcome if you have another way for resolve this.

**Comment by Hiws** [comment]
> I use a main.scss file that imported into my components

So you're importing all the CSS into every single component?

Try importing it only once, in your projects entry point.
And avoid placing it in a `scoped` style tag.

**Comment by darkomenx** [comment]
> So you're importing all the CSS into every single component?

I don't have a webpack configuration. And it seems to be not possible to import scss globally with vue.config.js.

I try this in the vue.config.js

```
css: {
  loaderOptions: {
    sass: {
      additionalData: `@import "./node_modules/bootstrap-vue/src/index.scss";`
    }
  }
}
```

but it generated this error :  

`PostCSS received undefined instead of CSS string`

- I also remove all the import declaration in my components
- I try with node-sass binary and dart-sass
- I also change key name options between scss, sass....
- I also remove scoped attribute on style components 


I can't load with vuejs a scss file for all the components.

**Comment by joshxyzhimself** [comment]
on: `PostCSS received undefined instead of CSS string`

in my experience if you are using webpack, and trying to use `sass-loader`, also add `sass`

```
yarn add sass sass-loader
```

```js
        {
          test: /\.(scss|css)$/,
          use: [
            { loader: 'style-loader' },
            { loader: 'css-loader', options: { importLoaders: 1 } },
            { loader: 'sass-loader' },
            { loader: 'postcss-loader' },
          ],
        },
```

**Comment by darkomenx** [comment]
Thank you @joshxyzhimself for your feedback.

I resolve my issue but unfortunately I don't know where problem from. I think that I was too many declaration in my project which cross between theirs. I resolve by substraction code and I don't import my global scss into all vuejs component.
